K8SSAND-1453 ⁃ Upgrade to cass-operator 1.10.3 (or later)

<!--

Feel free to ask questions on Discord at https://discord.com/invite/y4s64xC9 or in the forum at https://forum.k8ssandra.io/. 
-->

**What happened?**
DNS lookups for Cassandra pods does not work. See https://github.com/k8ssandra/cass-operator/issues/304.

**Did you expect to see something different?**
DNS resolution for Cassandra pods should work.

**How to reproduce it (as minimally and precisely as possible)**:
See https://github.com/k8ssandra/cass-operator/issues/304 for steps.

**Environment**

* K8ssandra Operator version:

    `v1.0.1`

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1453) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1453
┆priority: Medium

Upgrading cass-operator in main fails as follows:

```
$ go get "github.com/k8ssandra/cass-operator@v1.10.3"
go: downloading github.com/k8ssandra/cass-operator v1.10.3
go get: installing executables with 'go get' in module mode is deprecated.
	To adjust and download dependencies of the current module, use 'go get -d'.
	To install using requirements of the current module, use 'go install'.
	To install ignoring the current module, use 'go install' with a version,
	like 'go install example.com/cmd@latest'.
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.
# github.com/k8ssandra/cass-operator/pkg/httphelper
../../gocode/pkg/mod/github.com/k8ssandra/cass-operator@v1.10.3/pkg/httphelper/security.go:302:4: unknown field 'Handler' in struct literal of type "k8s.io/api/core/v1".Probe
../../gocode/pkg/mod/github.com/k8ssandra/cass-operator@v1.10.3/pkg/httphelper/security.go:305:25: container.LivenessProbe.Handler undefined (type *"k8s.io/api/core/v1".Probe has no field or method Handler)
../../gocode/pkg/mod/github.com/k8ssandra/cass-operator@v1.10.3/pkg/httphelper/security.go:306:25: container.LivenessProbe.Handler undefined (type *"k8s.io/api/core/v1".Probe has no field or method Handler)
../../gocode/pkg/mod/github.com/k8ssandra/cass-operator@v1.10.3/pkg/httphelper/security.go:307:25: container.LivenessProbe.Handler undefined (type *"k8s.io/api/core/v1".Probe has no field or method Handler)
../../gocode/pkg/mod/github.com/k8ssandra/cass-operator@v1.10.3/pkg/httphelper/security.go:313:4: unknown field 'Handler' in struct literal of type "k8s.io/api/core/v1".Probe
../../gocode/pkg/mod/github.com/k8ssandra/cass-operator@v1.10.3/pkg/httphelper/security.go:316:26: container.ReadinessProbe.Handler undefined (type *"k8s.io/api/core/v1".Probe has no field or method Handler)
../../gocode/pkg/mod/github.com/k8ssandra/cass-operator@v1.10.3/pkg/httphelper/security.go:317:26: container.ReadinessProbe.Handler undefined (type *"k8s.io/api/core/v1".Probe has no field or method Handler)
../../gocode/pkg/mod/github.com/k8ssandra/cass-operator@v1.10.3/pkg/httphelper/security.go:318:26: container.ReadinessProbe.Handler undefined (type *"k8s.io/api/core/v1".Probe has no field or method Handler)
```
This is because k8ssandra-operator depends on `k8s.io/kubernetes v1.23.4` in main whereas cass-operator v1.10.3 depends on `k8s.io/kubernetes v1.22.2`. cass-operator master branch is on `k8s.io/kubernetes v1.23.4`. 

We can upgrade to cass-operator v1.10.3 on the release/1.0 branch fine so that the upgrade can be included in k8ssandra-operator v1.0.2.

In the main branch we will need to upgrade to a more recent commit of cass-operator.

--- a/CHANGELOG/CHANGELOG-1.0.md
+++ b/CHANGELOG/CHANGELOG-1.0.md
@@ -14,6 +14,8 @@ Changelog for the K8ssandra Operator, new PRs should update the `unreleased` sec
 When cutting a new release, update the `unreleased` heading to the tag being generated and date, like `## vX.Y.Z - YYYY-MM-DD` and create a new placeholder section for  `unreleased` entries.
 
 ## unrelease
+* [CHANGE] [#520](https://github.com/k8ssandra/k8ssandra-operator/issues/520) Upgrade 
+  to cass-operator v1.10.3
 * [BUGFIX] [#503](https://github.com/k8ssandra/k8ssandra-operator/issues/503) Cannot 
   create Cassandra 4 multi-DC cluster when DCs have different sizes
 

--- a/config/cass-operator/cluster-scoped/kustomization.yaml
+++ b/config/cass-operator/cluster-scoped/kustomization.yaml
@@ -2,8 +2,8 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
 resources:
-- github.com/k8ssandra/cass-operator/config/deployments/cluster?ref=v1.10.0
+- github.com/k8ssandra/cass-operator/config/deployments/cluster?ref=v1.10.3
 
 images:
   - name: k8ssandra/cass-operator
-    newTag: v1.10.0
+    newTag: v1.10.3

--- a/config/cass-operator/ns-scoped/kustomization.yaml
+++ b/config/cass-operator/ns-scoped/kustomization.yaml
@@ -2,8 +2,8 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 
 resources:
-- github.com/k8ssandra/cass-operator/config/deployments/default?ref=v1.10.0
+- github.com/k8ssandra/cass-operator/config/deployments/default?ref=v1.10.3
 
 images:
   - name: k8ssandra/cass-operator
-    newTag: v1.10.0
+    newTag: v1.10.3

--- a/go.mod
+++ b/go.mod
@@ -17,7 +17,7 @@ require (
 	github.com/golang/snappy v0.0.4 // indirect
 	github.com/google/uuid v1.2.0
 	github.com/gruntwork-io/terratest v0.37.7
-	github.com/k8ssandra/cass-operator v1.10.0
+	github.com/k8ssandra/cass-operator v1.10.3
 	github.com/k8ssandra/reaper-client-go v0.3.1-0.20220114183114-6923e077c4f5
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.52.1

--- a/pkg/test/testenv.go
+++ b/pkg/test/testenv.go
@@ -43,7 +43,7 @@ import (
 const (
 	clustersToCreate          = 3
 	clusterProtoName          = "cluster-%d"
-	cassOperatorVersion       = "v1.10.0"
+	cassOperatorVersion       = "v1.10.3"
 	prometheusOperatorVersion = "v0.9.0"
 )
 

